src/pages/scan/scan.ts

```diff
@@ -173,5 +175,5 @@ export async function parseCustomQRCodeData(
       }
     }
   }
-  return null;
+  return matchOne;
```

Comment:
I'm not a fan of this to be honest. object|true|false, especially without any place checking for explicit `=== true` and `=== false`, feels quite confusing. 

And in fact, the `if (scannedData && !isRecord(scannedData))` doesn't look like something that would do the right thing:

- it won't run if `null` is returned (that's falsy)`
- it won't run if `false is returned (that's falsy as well)`
- it won't run if an object is returned

so in the end it'sc hecking for `scannedData === true`...

IMHO a much cleaner way would be to return a tuple or object containing two separate fields:

- a bool indicating whether some custom code was matched
- the IndicoParticipant (if you got something from indico) or null (ir nothing was matched or indico returned nothing)